src/main/java/com/mraof/minestuck/item/weapon/MusicPlayerWeapon.java

```diff
@@ -0,0 +1,236 @@
+		tag.putBoolean("HasCassette",
+				!getItemHandler(stack).getStackInSlot(0).isEmpty());
+		return tag;
+	}
+	
+	@SubscribeEvent
+	public static void playerTick(TickEvent.PlayerTickEvent tickEvent)
+	{
+		if(tickEvent.side.isServer() && tickEvent.phase == TickEvent.Phase.END)
+		{
+			Player player = tickEvent.player;
+			IMusicPlaying musicPlayingCap = getMusicPlaying(player);
+			
+			if(!(player.getItemInHand(InteractionHand.MAIN_HAND) == musicPlayingCap.getCurrentMusicPlayer() ||
+					player.getItemInHand(InteractionHand.OFF_HAND) == musicPlayingCap.getCurrentMusicPlayer()) ||
+					musicPlayingCap.getCurrentMusicPlayer().isEmpty())
+			{
+				if(musicPlayingCap.getCassetteType() != EnumCassetteType.NONE)
+				{ //If the Cassette player isn't in hand and is playing music, stop it
+					MusicPlayerPacket packet = MusicPlayerPacket.createPacket(player, EnumCassetteType.NONE, 0, 0);
+					musicPlayingCap.setMusicPlaying(ItemStack.EMPTY, EnumCassetteType.NONE);
+					MSPacketHandler.sendToTrackingAndSelf(packet, player);
+				}
+			}
+			
+			if(player.level.getGameTime() % 50 == 0 && musicPlayingCap.getCassetteType() != EnumCassetteType.NONE)
+			{
+				EnumCassetteType.EffectContainer effectContainer = musicPlayingCap.getCassetteType().getEffectContainer();
+				if(!effectContainer.onHit()) //Apply the cassette buff every 50 ticks, if there is any
+					player.addEffect(effectContainer.effect().get());
+			}
+		}
+	}
+	
+	@Override
+	public boolean hurtEnemy(ItemStack stack, LivingEntity target, LivingEntity attacker)
+	{
+		IMusicPlaying musicPlaying = getMusicPlaying(attacker);
+		if(musicPlaying.getCassetteType() != EnumCassetteType.NONE)
+		{
+			Random r = attacker.level.getRandom();
+			
+			double attackerLuckValue;
+			double targetLuckValue;
+			
+			EnumCassetteType.EffectContainer effectContainer = musicPlaying.getCassetteType().getEffectContainer();
+			AttributeInstance attackerLuck = attacker.getAttribute(Attributes.LUCK);
+			AttributeInstance targetLuck = target.getAttribute(Attributes.LUCK);
+			double chanceToHit = effectContainer.applyingChance();
+			
+			if(attackerLuck != null)
+				attackerLuckValue = attackerLuck.getValue();
+			else
+				attackerLuckValue = 0.0D;
+			
+			if(targetLuck != null)
+				targetLuckValue = targetLuck.getValue();
+			else
+				targetLuckValue = 0.0D;
+			//Compare the luck of the target with the user's
+			chanceToHit = chanceToHit + (attackerLuckValue / 10) - (targetLuckValue / 10);
+			
+			if(chanceToHit > r.nextFloat() && effectContainer.onHit())
+			{
+				target.addEffect(effectContainer.effect().get());
+			}
+		}
+		return super.hurtEnemy(stack, target, attacker);
+	}
+	
+	public static boolean hasCassette(ItemStack stack)
+	{
+		if(!stack.hasTag() || !stack.getTag().contains("HasCasette"))
+			return !getItemHandler(stack).getStackInSlot(0).isEmpty();
+		else return stack.getTag().getBoolean("HasCassette");
+	}
```

Comment:
Could be moved so that it is side-by-side with the corresponding function `getShareTag()`